src/repl.jl

```diff
@@ -220,14 +220,28 @@ end
 function LineEdit.complete_line(c::DebugCompletionProvider, s)
     partial = REPL.beforecursor(s.input_buffer)
     full = LineEdit.input_string(s)
-    ret, range, should_complete = completions(c, full, lastindex(partial))
-    return unique!(map(REPLCompletions.completion_text, ret)), partial[range], should_complete
+
+    ret, range, should_complete = completions(c, full, partial)
+    return ret, partial[range], should_complete
 end
 
 function completions(c::DebugCompletionProvider, full, partial)
-    mod = moduleof(c.state.frame)
-    ret, range, should_complete = REPLCompletions.completions(full, partial, mod)
+    frame = c.state.frame
+
+    # repl backend completions
+    comps, range, should_complete = REPLCompletions.completions(full, lastindex(partial), moduleof(frame))
+    ret = map(REPLCompletions.completion_text, comps) |> unique!
+
+    # local completions
+    vars = filter!(locals(frame)) do v
+        # xref: https://github.com/JunoLab/Atom.jl/blob/master/src/debugger/stepper.jl#L411-L421
+        if v.name == Symbol("#self") && (v.value isa Type || sizeof(v.value) == 0)
```

Comment:
we may remove the latter filter conditions here.

Reply:
We have that in JuliaInterpreter as well but I am not sure I see the reason for it. I think it can be removed here anyway.

Reply:
hm, maybe @timholy knows that ?

Reply:
ops, sorry I found those functions aren't defined, thus I suspected you just forgot to import them.
Do you want to still export them ?

Reply:
The `#self` is probably sufficient. It's not always there (I can't remember where I recently encountered such a situation), but it should be sufficient. Assuming I started that trend I was probably just being paranoid.